x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/QueryStringFunction.java

```diff
@@ -61,27 +73,18 @@ public String functionName() {
     }
 
     @Override
-    public Query asQuery() {
-        Object queryAsObject = query().fold();
-        if (queryAsObject instanceof BytesRef queryAsBytesRef) {
-            return new QueryStringQuery(source(), queryAsBytesRef.utf8ToString(), Map.of(), null);
-        } else {
-            throw new IllegalArgumentException("Query in QSTR needs to be resolved to a string");
-        }
+    public Query asQuery(String queryText) {
+        return new QueryStringQuery(source(), queryText, Map.of(), null);
```

Comment:
Move the build of QueryStringQuery into EsqlExpressionTranslators, it is a central place to build pushdown queries.

Reply:
Done in 06f2eafd9177a8d2d8c80300072039ac9e79407e